File: Task 3/Java/app/src/main/java/com/example/recyclerviewsqlitedemo/MainActivity.java
package com.example.recyclerviewsqlitedemo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    EditText et_name, et_email, et_age;
    Button btn_insert, btn_view;
    DatabaseHelper databaseHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        et_name = findViewById(R.id.et_name);
        et_email = findViewById(R.id.et_email);
        et_age = findViewById(R.id.et_age);
        btn_insert = findViewById(R.id.btn_insert);
        btn_view = findViewById(R.id.btn_view);

        databaseHelper = new DatabaseHelper(this);
        
        btn_view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, UserList.class));
            }
        });

        btn_insert.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = et_name.getText().toString();
                String email = et_email.getText().toString();
                String age = et_age.getText().toString();

                Boolean checkInsertTable = databaseHelper.insertUserData(name, email, age);
                if (checkInsertTable) {
                    Toast.makeText(MainActivity.this, "New Entry Inserted", Toast.LENGTH_SHORT).show();
                    et_name.setText("");
                    et_email.setText("");
                    et_age.setText("");
                }
                else {
                    Toast.makeText(MainActivity.this, "New Entry Not Inserted", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: Task 3/Kotlin/app/src/main/java/com/example/recyclerviewsqlitedemo/MainActivity.kt
package com.example.recyclerviewsqlitedemo

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    private lateinit var databaseHelper: DatabaseHelper
    private lateinit var userAdapter: UserAdapter
    private lateinit var recyclerView: RecyclerView
    private lateinit var btn_add_user: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        databaseHelper = DatabaseHelper(this)
        recyclerView = findViewById(R.id.recycler_view)
        btn_add_user = findViewById(R.id.btn_add_user)

        btn_add_user.setOnClickListener {
            showAddUserDialog()
        }

        setupRecyclerView()
        loadUserData()
    }

    private fun setupRecyclerView() {
        recyclerView.layoutManager = LinearLayoutManager(this)
    }

    private fun loadUserData() {
        val users = databaseHelper.getAllUsers()
        userAdapter = UserAdapter(users)
        recyclerView.adapter = userAdapter
    }

    private fun showAddUserDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Add User")

        val inflater = layoutInflater
        val dialogLayout = inflater.inflate(R.layout.dialog_add_user, null)
        builder.setView(dialogLayout)

        val et_name = dialogLayout.findViewById<EditText>(R.id.et_name)
        val et_age = dialogLayout.findViewById<EditText>(R.id.et_age)

        builder.setPositiveButtonIcon("Add") { dialog, _ ->
            val name = et_name.text.toString()
            val age = et_age.text.toString().toIntOrNull()

            if (name.isNotEmpty() && age != null) {
                databaseHelper.addUser(name, age)
                loadUserData()
                Toast.makeText(this, "User added", Toast.LENGTH_SHORT).show()
            }
            else {
                Toast.makeText(this, "Invalid input", Toast.LENGTH_SHORT).show()
            }
            dialog.dismiss()
        }

        builder.setNavigationButton("Cancel") { dialog, _ ->
            dialog.dismiss()
        }

        builder.show()
    }
}